### PR TITLE
EUI-9106 Hearing CR-84 Incorrect field used when comparing party

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/EUI-9106

### Change description ###
Hearing CR84 - Party name - Incorrect field is used when comparing this between HMC and SHV values

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.spec.ts [hunk 1]
```diff
@@ -469,6 +469,18 @@ describe('HearingEditSummaryComponent', () => {
...
   });
 
+  it('should partyDetailsChangesRequired return true if party type changed', () => {
+    spyOn(store, 'select').and.returnValue(of(initialState.hearings.hearingValues));
+    component.serviceHearingValuesModel = _.cloneDeep(initialState.hearings.hearingValues.serviceHearingValuesModel);
+    component.serviceHearingValuesModel = {
+      ...component.serviceHearingValuesModel,
+      parties: initialState.hearings.hearingRequest.hearingRequestMainModel.partyDetails
+    };
+    component.serviceHearingValuesModel.parties[0].partyType = PartyType.ORG;
```
Comment (Daniel-Lam) [inline src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.spec.ts:479]: @johnbenjamin-hmcts - is it guaranteed there will always be at least one item in `parties` array?

File: src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts [hunk 2]
```diff
@@ -377,11 +377,10 @@ export class HearingEditSummaryComponent extends RequestHearingPageFlow implemen
     // and return true if there are any changes in the party name of party type
     for (const partySHV of partiesSHV) {
       const party = partiesHMC.find((partyHMC) => partyHMC.partyID === partySHV.partyID);
-      if ((party.partyName !== partySHV.partyName) ||
-        (party.individualDetails?.title !== partySHV.individualDetails?.title) ||
-        (party.individualDetails?.firstName !== partySHV.individualDetails?.firstName) ||
-        (party.individualDetails?.lastName !== partySHV.individualDetails?.lastName) ||
-        (party.partyType !== partySHV.partyType)) {
+      if (party.partyType !== partySHV.partyType) {
```
Comment (Daniel-Lam) [inline src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts:380]: @johnbenjamin-hmcts - maybe these could be joined together with `||`, as they both return `true`

**Review by udaydenduluri33 — APPROVED** [review]

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - some very minor observations but otherwise OK

File: src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts]: Thank you @Daniel-Lam, I have updated the code now.

**Review by Daniel-Lam — APPROVED** [review]

**Review by udaydenduluri33 — APPROVED** [review]

**Pull request merged**